src/bin/svg2png/svg2png.cpp

```diff
@@ -25,180 +25,397 @@
 #include <thorvg.h>
 #include <vector>
 #include "lodepng.h"
+#include <dirent.h>
+#include <unistd.h>
+#include <limits.h>
 
 using namespace std;
 
+#define FLAG_DISABLE_SUBDIRS (1)
+#define FLAG_USE_ABGR8888_COLORSPACE (1 << 1)
+#define FLAG_DO_NOT_SAVE_PNG (1 << 2)
 
 struct PngBuilder
 {
-    void build(const std::string &fileName , const uint32_t width, const uint32_t height, uint32_t *buffer)
+    void build(const string& fileName, const uint32_t width, const uint32_t height, uint32_t* buffer, bool altColorspace)
     {
-        std::vector<unsigned char> image;
-        image.resize(width * height * 4);
-        for (unsigned y = 0; y < height; y++) {
-            for (unsigned x = 0; x < width; x++) {
-                uint32_t n = buffer[ y * width + x ];
-                image[4 * width * y + 4 * x + 0] = (n >> 16) & 0xff;
-                image[4 * width * y + 4 * x + 1] = (n >> 8) & 0xff;
-                image[4 * width * y + 4 * x + 2] = n & 0xff;
-                image[4 * width * y + 4 * x + 3] = (n >> 24) & 0xff;
+        unsigned error = 0;
+        if (!altColorspace) {
+            //Used ARGB8888 so have to move pixels now
+            vector<unsigned char> image;
+            image.resize(width * height * 4);
+            for (unsigned y = 0; y < height; y++) {
+                for (unsigned x = 0; x < width; x++) {
+                    uint32_t n = buffer[y * width + x];
+                    image[4 * width * y + 4 * x + 0] = (n >> 16) & 0xff;
+                    image[4 * width * y + 4 * x + 1] = (n >> 8) & 0xff;
+                    image[4 * width * y + 4 * x + 2] = n & 0xff;
+                    image[4 * width * y + 4 * x + 3] = (n >> 24) & 0xff;
+                }
             }
+            error = lodepng::encode(fileName, image, width, height);
+
+        } else {
+            //Used ABGR8888 so no need to move pixels
+            error = lodepng::encode(fileName, (const unsigned char*) buffer, width, height);
         }
```

Comment:
i have a question..
If the user changes the colorspace using -c, shouldn't the color of the output also change?

Reply:
With and without -c flag it should look the same.
In the primary version of svg2png, ARGB8888 colorspace was used. As lodepng needs ABRG, color pixels was moved from ARGB to ABGR before calling `lodepng::encode`. I added -c that uses ABGR8888 at the beginning, so there is no need for moving pixels.
It can be tested with something that doesnt work correctly now, eg. png images. Png image can be embed in svg image and than colors in generated image will be changed.